main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapActivity.java

```diff
@@ -571,9 +571,8 @@ private String calculateTitle() {
     private void refreshListChooser() {
         //try to get count of visible caches
         int visibleCaches = 0;
-        final Viewport viewport = mapFragment.getViewport();
-        if (mapFragment != null && viewport != null && viewModel != null) {
-            visibleCaches = viewModel.caches.readWithResult(viewport::count);
+        if (mapFragment != null && mapFragment.getViewport() != null && viewModel != null) {
+            visibleCaches = viewModel.caches.readWithResult(mapFragment.getViewport()::count);
```

Comment:
out of curiosity: what is the difference between `mapFragment.getViewport()::count` and `viewport::count` with `viewport` being set to `mapFragment.getViewport()`?

Reply:
There isn't any. 

mapFregment may be null, so that needs a null check before getViewport(). And the variable is used only once, so no point in creating it.

Reply:
Ah, you're right, `mapFragment` could be null, crashing in line 574 then - had missed that. Thanks for clarifying and for the fix.